### PR TITLE
Swagger Updates

### PR DESCRIPTION
**Pull request by abdelrahman-mos**

### CODE REVIEW

File: src/swagger.ts
Comment (Deadreyo) [inline src/swagger.ts:1]: Participant schema is outdated.

Comment (Deadreyo) [inline src/swagger.ts:1]: Log schema is outdated.

Comment (Deadreyo) [inline src/swagger.ts:1]: Required: When using dev mode, a server for localhost should be the default instead of the production server.
image

Comment (Deadreyo) [inline src/swagger.ts:1]: Search bar gives an error.

**Review by nightknighto — CHANGES_REQUESTED** [review]
Impressive work so far! Just a few notes:

File: src/swagger.ts
Comment (Deadreyo) [inline src/swagger.ts:1]: Definitions for schemas should be automatically generated. I have 2 ideas in mind for this:
1- There is a library that can convert TypeScript types into JSON Schemas, and we try to insert that into swagger. We can look up more approaches similar to this.
2- Fields can be extracted from mongoose models. We can construct an object from this and insert into swagger.

**Pull request merged**